src/features/antora/antoraSupport.ts

```diff
@@ -0,0 +1,86 @@
+import vscode, { CancellationTokenSource, Uri } from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import * as path from 'path'
+import AntoraCompletionProvider from './antoraCompletionProvider'
+import { disposeAll } from '../../util/dispose'
+
+export class AntoraSupportManager {
+  private readonly _disposables: vscode.Disposable[] = []
+
+  public constructor (private readonly context: vscode.ExtensionContext) {
+    this.context = context
+    this._disposables.push(vscode.languages.registerCompletionItemProvider(
+      {
+        language: 'asciidoc',
+        scheme: 'file',
+      },
+      new AntoraCompletionProvider(),
+      '{'
+    ))
+    this._disposables.push(vscode.workspace.onDidOpenTextDocument(async (textDocument) => {
+      const workspaceConfiguration = vscode.workspace.getConfiguration('asciidoc', null)
+      // look for Antora settings in workspaceState
+      const workspaceState: vscode.Memento = this.context.workspaceState
+      const isEnableAntoraSupportSettingDefined = await workspaceState.get('antoraSupportSetting')
+      if (isEnableAntoraSupportSettingDefined === true) {
+        if (workspaceConfiguration.get('antora.enableAntoraSupport') === true) {
+          this.dispose()
+        } else {
+          // no op
+        }
```

Comment:
If the Antora support is enabled we `dispose` something? I believe the idea was to register the completionItemProvider above only when the Antora support is enabled?

Reply:
yes you 're right, I've mixed everything up by changing the logic